src/Groups/homomorphisms.jl

```diff
@@ -386,22 +386,27 @@ function hom(x::GAPGroupElem{AutomorphismGroup{T}}) where T <: GAPGroup
   return _hom_from_gap_map(G, G, x.X)
 end
 
-(f::GAPGroupElem{AutomorphismGroup{T}})(x::GAPGroupElem{T}) where T <: GAPGroup = apply_automorphism(f, x)
-Base.:^(x::GAPGroupElem{T},f::GAPGroupElem{AutomorphismGroup{T}}) where T <: GAPGroup = apply_automorphism(f, x)
+(f::GAPGroupElem{AutomorphismGroup{T}})(x::GAPGroupElem) where T <: GAPGroup = apply_automorphism(f, x)
+Base.:^(x::GAPGroupElem,f::GAPGroupElem{AutomorphismGroup{T}}) where T <: GAPGroup = apply_automorphism(f, x)
+Base.:^(f::GAPGroupElem{AutomorphismGroup{T}},g::GAPGroupElem{AutomorphismGroup{T}}) where T <: GAPGroup = g^-1*f*g
 
 function (A::AutomorphismGroup{T})(f::GAPGroupHomomorphism{T,T}) where T <: GAPGroup
    @assert domain(f)==A.G && codomain(f)==A.G "f not in A"
    @assert isbijective(f) "f not in A"
    return group_element(A, f.map)
 end
 
-function apply_automorphism(f::GAPGroupElem{AutomorphismGroup{T}}, x::GAPGroupElem{T}, check=true) where T <: GAPGroup
+function apply_automorphism(f::GAPGroupElem{AutomorphismGroup{T}}, x::GAPGroupElem, check=true) where T <: GAPGroup
   A = parent(f)
   G = parent(x)
   if check
-    @assert A.G == G || GAP.Globals.IN(x.X, A.G.X) "Not in the domain of f!"
+    @assert A.G == G || GAP.Globals.IN(x.X, A.G.X) "Not in the domain of f!"      #TODO Do we really need the IN check?
+  end
+  if x isa MatrixGroupElem
+     return MatrixGroupElem(G,GAP.Globals.Image(f.X,x.X))
```

Comment:
I don't understand why this is necessary, can you explain? It is also not covered by a test -- perhaps you can write a code example which triggers this, and then tell us which error this code example produces if one removes this special case.

Reply:
Do you mean the assertion or the `if isa MatrixGroupElem` ?
The first one is to avoid to compute the automorphism on an element that is not in its domain, that creates a long GAP error.
For example: the following piece of code

> A = automorphism_group( alternating_group(5)); f = rand(A);
> x = cperm([1,3,5,4]);
> Oscar.apply_automorphism(f,x, false);
produces the error
> ERROR: Error thrown by GAP: Error, <elm> must be an element of Source(<map>)
The distinction between `MatrixGroupElem` and the other types is that if `x` is a matrix, then its type is `MatrixGroupElem{T}`, but the constructor is simply `MatrixGroupElem` (without the parameter), so the line
> return typeof(x)(G, GAP.Globals.Image(f.X,x.X))
does not work.

Reply:
That's arguably a bug (or misfeature) in your code: there *should* be a `MatrixGroupElem{T}` constructor. Please take a look at https://github.com/oscar-system/Oscar.jl/pull/267 where I resolved a very similar issue

Reply:
I wrote the parametric constructors for `MatrixGroup` and `MatrixGroupElem`. What if the arguments of the constructors have a different type from the parameters? Or suptype/subtype? I mean, what should happen if I have something like `MatrixGroupElem{RE, T}(G, x::S)`, where the `S` is a subtype/suptype of `T` ? Do I return a variable of type `MatrixGroupElem{RE,S}`? Or a variable of type `MatrixGroupElem{RE,T}`? Or an ERROR message?
For the moment I lean towards the first option.